core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -353,38 +353,60 @@ private[spark] class MesosClusterScheduler(
     }
   }
 
-  private def buildDriverCommand(desc: MesosDriverDescription): CommandInfo = {
-    val appJar = CommandInfo.URI.newBuilder()
-      .setValue(desc.jarUrl.stripPrefix("file:").stripPrefix("local:")).build()
-    val builder = CommandInfo.newBuilder().addUris(appJar)
-    val entries = conf.getOption("spark.executor.extraLibraryPath")
-      .map(path => Seq(path) ++ desc.command.libraryPathEntries)
-      .getOrElse(desc.command.libraryPathEntries)
-
-    val prefixEnv = if (!entries.isEmpty) {
-      Utils.libraryPathEnvPrefix(entries)
-    } else {
-      ""
+  private def getDriverExecutorURI(desc: MesosDriverDescription) = {
+    desc.schedulerProperties.get("spark.executor.uri")
+      .orElse(desc.command.environment.get("SPARK_EXECUTOR_URI"))
+  }
+
+  private def getDriverEnvironment(desc: MesosDriverDescription): Environment = {
+    val env = {
+      val executorOpts = desc.schedulerProperties.map { case (k, v) => s"-D$k=$v" }.mkString(" ")
+      val executorEnv = Map("SPARK_EXECUTOR_OPTS" -> executorOpts)
+
+      val prefix = "spark.mesos.env."
```

Comment:
> From a UX experience and name consistency view i would expect something like that.

Yea I went back and forth on this.  I'm fine with spark.mesos.driverEnv  Does that work?

> also in client mode you do not need that so you will need to ignore it.

Do you mean there needs to be some code change to ignore it?  The only handling code is in the cluster dispatcher, so it's effectively ignored in client mode.